test/shared/src/test/scala/monocle/GetterSpec.scala

```diff
@@ -57,4 +64,26 @@ class GetterSpec extends MonocleSuite {
     i.exist(_ > 5)(Bar(3)) shouldEqual false
   }
 
+  test("zip") {
+    val length = Getter[String, Int](_.length)
+    val upper = Getter[String, String](_.toUpperCase)
+    length.zip(upper).get("helloworld") shouldEqual((10, "HELLOWORLD"))
+  }
+
+  test("unzip") {
+    val lengthAndUpper = Getter[String, (Int, String)](s => s.length -> s.toUpperCase)
+    val (length, upper) = lengthAndUpper.unzip[Int, String]
+    length.get("helloworld") shouldEqual 10
+    upper.get("helloworld") shouldEqual "HELLOWORLD"
+
+  }
+
+  test("zip/unzip roundtrip") {
+    val length = Getter[String, Int](_.length)
+    val upper = Getter[String, String](_.toUpperCase)
+
+    val (length1, upper1) = length.zip(upper).unzip[Int, String]
```

Comment:
arf ok, maybe we should revert back to your first commit if we can't get the type inference to work properly